Src/BootCamp.Chapter/ToggleableGridJagged.cs

```diff
@@ -1,18 +1,52 @@
-﻿using System;
+﻿ using System;
+using System.Text; 
 
 namespace BootCamp.Chapter
 {
     public class ToggleableGridJagged : IToggleableGrid
     {
         private readonly IGridClearer _gridClearer;
+        private readonly bool[][] _toggle;
 
         public ToggleableGridJagged(bool[][] toggles, IGridClearer gridClearer)
         {
+            _toggle = toggles;
             _gridClearer = gridClearer;
         }
 
         public void Toggle(int x, int y)
         {
+            _toggle[x][y] = !_toggle[x][y];
+            StringBuilder sb = new StringBuilder();            
+            CheckInput(x, y);
+            for(int i = 0; i < _toggle.GetLength(0); i++)
+            {
+                for (int j = 0; j < _toggle[i].Length; j++)
+                {
+                    if (_toggle[i][j])
+                    {
+                        sb.Append("■");
+                    }
+                    else
+                    {
+                        sb.Append(" ");
+                    }
+                }
+                sb.Append(Environment.NewLine);
+            }           
+            var str = sb.ToString().TrimEnd("\r\n".ToCharArray());
```

Comment:
Split to a function of its own

Reply:
So the function will encapsulate sb- it's a low level detail

Reply:
Avoid mixing high level functions with implementation of other functions.

Reply:
Sorry for late response.
For this issue, I regard 2D array as special jagged array and define DisplayGrid class to reuse.
Also spend some time to solve the null reference exception about how to make 2D array works.
Through this I could handle 2D array initialization